src/Proxy.hs

```diff
@@ -1999,22 +2016,22 @@ getNextPaydayR =
       sendResponse $ toJSON timestampObject
   where
     doGetParameters = do
-      rewardStatus <- withLastFinalBlockHash Nothing getRewardStatus
-      return $ do
-        response <- rewardStatus
-        utc <- parseEither
-          ( withObject "Best finalized block" $ \v -> do
-              v AE..: "nextPaydayTime"
-          ) $ grpcResponseVal response
-        Right $ GRPCResponse (grpcHeaders response) utc
+      rewardStatusRes <- getTokenomicsInfo LastFinal
+      case getResponseValueAndHeaders rewardStatusRes of
+        Left errRes -> return errRes
+        Right (Left err, hds) -> return $ StatusOk $ GRPCResponse hds $ Left err
+        Right (Right rewardStatus, hds) -> do
+          case rewardStatus of
+            RewardStatusV0{} ->
+              return $ StatusOk $ GRPCResponse hds $ Left "Received unexpected 'RewardStatusV0' variant in response payload, expected 'RewardStatusV1'."
```

Comment:
Won't this be then mapped to `InternalServerError` by the `runGRPC` call?

That does not seem appropriate since this is an invalid request, the client requested next payday when the protocol does not support it.

Reply:
I changed this now. I am not sure I am happy with this, since the specified invalid request error will be returned if the protocol buffer error conversion fails. Maybe we can discuss if it is necessary to change the return type of the client monad to allow for further distinguishing between errors.

```
    runGRPCWithCustomError
      (Just (InvariantError, Just badRequest400, Just RequestInvalid, Just $ EMErrorResponse $ InvalidArgs []))
```

Reply:
The error message is still wrong. There is no a priori reason this should expect a V1. So it is not "unexpected".

It's just that the state of hte chain is such that they were received.